src/StreamJsonRpc.Tests/MessagePackFormatter.cs

```diff
@@ -21,13 +25,60 @@ namespace StreamJsonRpc
     /// The README on that project site describes use cases and its performance compared to alternative
     /// .NET MessagePack implementations and this one appears to be the best by far.
     /// </remarks>
-    public class MessagePackFormatter : IJsonRpcMessageFormatter
+    public class MessagePackFormatter : IJsonRpcMessageFormatter, IJsonRpcInstanceContainer
     {
+        /// <summary>
+        /// Special method name for progress notification.
+        /// </summary>
+        private const string ProgressRequestSpecialMethod = "$/progress";
+
         /// <summary>
```

Comment:
I see a lot of common fields between this and the JsonMessageFormatter class. Can we move these into your helper class?